extensions/builder.md

```diff
@@ -0,0 +1,180 @@
+# Builder Specification <!-- omit in toc -->
+
+This document specified the artifact format for a builder.
+
+A builder is an OCI image that provides a distributable build environment.
+
+A [platform][platform-spec] supporting the builder extension specification SHOULD allow users to configure the build environment with a provided builder.
+
+## Table of Contents <!-- omit in toc -->
+- [General Requirements](#general-requirements)
+  - [Builder API Version](#builder-api-version)
+  - [Filesystem](#filesystem)
+  - [Environment Variables](#environment-variables)
+  - [Labels](#labels)
+- [Data Format](#data-format)
+  - [Labels](#labels-1)
+    - [`io.buildpacks.builder.metadata` (JSON)](#iobuildpacksbuildermetadata-json)
+    - [`io.buildpacks.buildpack.order` (JSON)](#iobuildpacksbuildpackorder-json)
+    - [`io.buildpacks.buildpack.layers` (JSON)](#iobuildpacksbuildpacklayers-json)
+    - [`io.buildpacks.lifecycle.apis` (JSON)](#iobuildpackslifecycleapis-json)
+
+## General Requirements
+A **builder** is an image that MUST contain an implementation of the lifecycle, and build-time environment, and MAY contain buildpacks. Platforms SHOULD use builders to ease the build process.
+
+### Builder API Version
+This document specifies Builder API version `0.1`.
+
+Builder API versions:
+- MUST be in form `<major>.<minor>` or `<major>`, where `<major>` is equivalent to `<major>.0`
+- When `<major>` is greater than `0` increments to `<minor>` SHALL exclusively indicate additive changes
+
+### Filesystem
+A builder MUST have the following directories/files:
+- `/cnb/order.toml` &rarr; As defined in the [platform specification][order-toml-spec]
+- `/cnb/stack.toml` &rarr; As defined in the [platform specification][stack-toml-spec]
+- `/cnb/lifecycle/<lifecycle binaries>` &rarr; An implementation of the lifecycle, which contains the required lifecycle binaries for [building images][lifecycle-for-build].
+
+In addition, every buildpack blob contained on a builder MUST be stored at the following file path:
+- `<CNB_BUILDPACKS_DIR>/<buildpack ID>/<buildpack version>/`
+
+If the buildpack ID contains a `/`, it MUST be replaced with `_` in the directory name.
+
+All specified files and directories are writeable by the build environment's User. 
```

Comment:
```suggestion
The `CNB_APP_DIR` and `CNB_LAYERS_DIR` SHOULD be writeable by the build environment's User.
```

Reply:
Suggested as SHOULD since I don't think it's required (the lifecycle will change permissions on those directories). Maybe we want it to be a MUST though?

Reply:
It would be nice to get the lifecycle out of the chowning business in my opinion, which we could do if it was gauranteed that this directory is writable by the build user.